cfme/common/vm.py

```diff
@@ -237,6 +239,25 @@ def compliant(self):
         else:
             raise ValueError("{} is not a known state for compliance".format(text))
 
+    @property
+    def console_handle(self):
+        '''
+        The basic algorithm for getting the consoles window handle is to get the
+        appliances window handle and then iterate through the window_handles till we find
+        one that is not the appliances window handle.   Once we find this check that it has
+        a canvas widget with a specific ID
+        '''
+        browser = self.appliance.browser.widgetastic
+        appliance_handle = browser.window_handle
+        cur_handles = browser.selenium.window_handles
+        logger.info("Current Window Handles:  {}".format(cur_handles))
+
+        for handle in cur_handles:
+            if handle != appliance_handle:
+                # XXX: Add code to verify the tab has the correct widget
```

Comment:
**Required**

Is this too complex to include in the scope here? Needs to be a FIXME, ideally at least some bare minimum check implemented.